addons/sale_stock/tests/test_create_perf.py

```diff
@@ -77,7 +77,7 @@ def test_empty_sales_orders_batch_creation_perf(self):
     def test_dummy_sales_orders_batch_creation_perf(self):
         """ Dummy SOlines (notes/sections) should not add any custom queries other than their insert"""
         # + 2 SOL (batched) insert
-        with self.assertQueryCount(admin=37):
+        with self.assertQueryCount(admin=39):
```

Comment:
Can you explain why there are 2 queries here even though it is the same partner twice?

Reply:
Up?

Reply:
I tried running these tests (with the original values for the query count) again locally to determine why 39 was required as a query count and the tests aren't failing, so I've dropped the changes from this commit. The EU countries are still queried during the test, so I suspect the performance has been improved somewhere else, allowing for this commit to no longer exceed the query count.

Reply:
You are actually doing 4 more queries everywhere, but the counters were out of date, so we see a difference here.
* `exists()` on `res.country.group` twice (in `ref()`)
```sql
select "res_country_group"."id" from "res_country_group" where "res_country_group".id in %s
```
* fetch of the `country_ids`
```sql
select res_country_res_country_group_rel.res_country_group_id, res_country_res_country_group_rel.res_country_id from res_country_res_country_group_rel, "res_country" where 1=1 and res_country_res_country_group_rel.res_country_group_id in %s and res_country_res_country_group_rel.res_country_id = res_country.id order by "res_country"."name"->>\'en_us\'
```
* `read()` on the `res.country`
```sql
select "res_country"."id", "res_country"."name"->>\'en_us\', "res_country"."code", "res_country"."address_format", "res_country"."address_view_id", "res_country"."currency_id", "res_country"."phone_code", "res_country"."name_position", "res_country"."vat_label"->>\'en_us\', "res_country"."state_required", "res_country"."zip_required", "res_country"."create_uid", "res_country"."create_date", "res_country"."write_uid", "res_country"."write_date" from "res_country" where ("res_country"."id" in %s)
```

Reply:
Thanks for figuring this one out :)